File: src/api/request.ts
import { invoke } from "@tauri-apps/api/core";
import { TRequest } from "../types";

export const addRequsetApi = async (requests: TRequest[]) => {
  try {
    const res = await invoke("add_request", {
      requests,
    });
    return res as string;
  } catch (e) {
    console.error("Faild to add request");
  }
};
export const getRequsetsApi = async (id: string) => {
  try {
    const res = await invoke("get_request_by_id", {
      requestId: id,
    });
    return res;
  } catch (e) {
    console.error("Faild to add request");
  }
};

export const updateRequestByIdApi = async (
  allRequestsId: string,
  requestId: string,
  updatedRequest: TRequest,
) => {
  try {
    const res = await invoke("update_request_by_id", {
      allRequestsId,
      requestId,
      updatedRequest,
    });
    return res;
  } catch (e) {
    console.error("Failed to update request", e);
    throw e;
  }
};

File: src-tauri/src/handlers/request.rs
use crate::datastore::request;
use crate::types::{AllRequests, Request};

#[tauri::command]
pub async fn add_request(requests: Vec<Request>, window: tauri::Window) -> Result<String, String> {
    println!("Handler - Received requests: {:?}", requests);
    request::add_request_db(requests, &window)
}

#[tauri::command]
pub async fn get_request_by_id(
    request_id: String,
    window: tauri::Window,
) -> Result<AllRequests, String> {
    request::get_request_by_id_db(request_id, &window)
}

#[tauri::command]
pub async fn update_request(
    all_requests: AllRequests,
    window: tauri::Window,
) -> Result<(), String> {
    request::update_request_db(all_requests, &window)
}

#[tauri::command]
pub async fn update_request_by_id(
    all_requests_id: String,
    request_id: String,
    updated_request: Request,
    window: tauri::Window,
) -> Result<(), String> {
    request::update_request_by_id_db(all_requests_id, request_id, updated_request, &window)
}
